apps/system/js/card.js

```diff
@@ -350,11 +340,132 @@
+        this.deltaX = 0;
+        this.deltaY = 0;
+        this.startTouchPosition = [evt.touches[0].pageX, evt.touches[0].pageY];
+        break;
+
+      case 'touchmove':
+        evt.stopPropagation();
+        this.deltaX = evt.touches[0].pageX - this.startTouchPosition[0];
+        this.deltaY = this._ease(
+          evt.touches[0].pageY - this.startTouchPosition[1],
+          this.manager.SWIPE_UP_THRESHOLD
+        );
+        verticalY = -1 * this.deltaY;
+        switch (this._dragPhase) {
+          case '':
+            if (verticalY > Math.abs(this.deltaX) &&
+                verticalY > this.SWIPE_WOBBLE_THRESHOLD) {
+              this._dragPhase = 'cross-slide';
+              // dont try and transition while dragging
+              this.contentNode.style.transition = 'transform 0s linear';
+              this.onCrossSlide(evt);
+            }
+            break;
+
+          case 'cross-slide':
+            this.onCrossSlide(evt);
+            break;
+        }
+        break;
+
+      case 'touchend':
+        this.deltaX = evt.changedTouches[0].pageX - this.startTouchPosition[0];
+        this.deltaY = this._ease(
+          evt.changedTouches[0].pageY - this.startTouchPosition[1],
+          this.manager.SWIPE_UP_THRESHOLD
+        );
+        if (Math.abs(this.deltaX) <= tapThreshold ||
+            Math.abs(this.deltaY) <= tapThreshold) {
+          this._resetY();
+          return;
+        }
+        verticalY = -1 * this.deltaY;
+        // cross-slide should be more up than across
+        if (verticalY > Math.abs(this.deltaX) &&
+            verticalY > this.manager.SWIPE_UP_THRESHOLD &&
+            this.app.killable()) {
+          // leave the card where it is if it will be destroyed
+          this.killApp();
+        } else {
+          // return it to vertical center
+          this._resetY();
+        }
+        break;
+    }
+  };
+
+  /**
+   * @memberOf Card.prototype
+   * @param {DOMEvent} evt
+   */
+  Card.prototype.onCrossSlide = function(evt) {
+    // move card up by the delta - the threshold
+    var offsetY = this.deltaY - this.SWIPE_WOBBLE_THRESHOLD;
+    this.contentNode.style.transform = 'translateY(' + offsetY + 'px)';
+  };
+
+  /**
+   * Ease for y-axis movement to damp start of the cross-slide
+   * @memberOf Card.prototype
+   * @param x {number} incremental value from 0 to max
+   * @param max {number}
+   */
+  Card.prototype._ease = function(x, max) {
+    var y;
+    var pt = Math.abs(x)/max;
+    var sign = x >= 0 ? 1 : -1;
+    var q1 = 0.25 - Math.pow(0.25, 1.675);
+    if (pt <= 0.25) {
+      // ease in at first
+      y = Math.pow(pt, 1.675);
+    } else {
+      // linear (minus offet)
+      y = pt - q1;
+    }
+    return sign * max * y;
+  };
+
+  /**
+   * @memberOf Card.prototype
+   */
+  Card.prototype._resetY = function(evt) {
+    // remove the inline transition so we transition per the stylesheet
+    this.contentNode.style.removeProperty('transition-property');
```

Comment:
I could not understand was the card dragging was so jumpy. Turns out we're setting `style.transition` when switching to cross sliding but removing `transition-propery` here which breaks the intended extra smoothness.

anyway doing a `removeProperty('transition')` here seems to fix it and brings back the transition when we don't kill the card.